pandas/core/series.py

```diff
@@ -2878,16 +2878,22 @@ def from_csv(cls, path, sep=',', parse_dates=True, header=None,
 
         return result
 
-    def to_csv(self, path=None, index=True, sep=",", na_rep='',
+    @deprecate_kwarg(old_arg_name='path', new_arg_name='path_or_buf')
+    def to_csv(self, path_or_buf=None, index=True, sep=",", na_rep='',
                float_format=None, header=False, index_label=None,
                mode='w', encoding=None, compression=None, date_format=None,
-               decimal='.'):
+               decimal='.', **kwargs):
```

Comment:
don't pass kwargs. rather list all of the args.

Reply:
Why not passing `kwargs` ? Otherwise each time we add something to `to_csv`, we will forget to add it here. We can just clearly say in the docstring here: "if you want to see all params, check DataFrame.to_csv"

Reply:
this is not a good pattern 
much better to have a consistent signature
defined

very unlikely to add new keywords anyhow
shared docs are the way to do this
and follow our pattern

kwargs are not

Reply:
As @dahlbaek said, and it might be a pity, but the kwarg order is not equal, so we can't just share. I agree with you in general, but I don't think it is worth to put much effort in this case to make it work (deprecate positional keywords etc)

Reply:
let’s just fix the args and conform to DataFrame.to_csv
otherwise this will go on forever

Reply:
Just changing the order is a backwards incompatible change, and needs to be deprecated first.

Reply:
of course but it’s better to have a 1 time break
than let this go on forever

Reply:
- Is there a standard way to deprecate ordering of positional arguments?
- Is there a standard way to deprecate default value of a keyword argument?
- Would it be ok to go for a one argument per line approach, as is done in [pandas.read_excel](https://github.com/dahlbaek/pandas/blob/master/pandas/io/excel.py#L282-L303)? I find this style much more readable, and changes are easier to track too.

Reply:
I don't think there is a similar example currently in the codebase. But I think doing a `*args, **kwargs` as you proposed in the top post, is the easiest way to go. You can then introspect the `args`, assign them manually to the correct keywords, and raise a DeprecationWarning if needed.